projects/packages/google-fonts-provider/src/class-google-fonts-provider.php

```diff
@@ -0,0 +1,343 @@
+ * @since $$next-version$$
+ */
+
+namespace Automattic\Jetpack\Fonts;
+
+if ( ! class_exists( '\WP_Webfonts_Provider' ) ) {
+	return;
+}
+
+/**
+ * Google Font Provider
+ */
+class Google_Fonts_Provider extends \WP_Webfonts_Provider {
+	/**
+	 * Font provider ID.
+	 *
+	 * @var string
+	 */
+	protected $id = 'google';
+
+	/**
+	 * The provider's root URL for retrieving font CSS.
+	 *
+	 * @var string
+	 */
+	protected $root_url = 'https://fonts.googleapis.com/css2';
+
+	/**
+	 * Gets cached CSS from a remote URL.
+	 *
+	 * @param string $id   An ID used to cache the styles.
+	 * @param string $url  The URL to fetch.
+	 * @param array  $args Optional. The arguments to pass to `wp_safe_remote_get()`.
+	 *                     Default empty array.
+	 * @return string The styles.
+	 */
+	protected function get_cached_remote_styles( $id, $url, array $args = array() ) {
+		$css = get_site_transient( $id );
+
+		// Get remote response and cache the CSS if it hasn't been cached already.
+		if ( false === $css ) {
+			$css = $this->get_remote_styles( $url, $args );
+
+			/*
+			* Early return if the request failed.
+			* Cache an empty string for 60 seconds to avoid bottlenecks.
+			*/
+			if ( empty( $css ) ) {
+				set_site_transient( $id, '', MINUTE_IN_SECONDS );
+				return '';
+			}
+
+			// Cache the CSS for a month.
+			set_site_transient( $id, $css, MONTH_IN_SECONDS );
+		}
+
+		return $css;
+	}
+
+	/**
+	 * Gets styles from the remote font service via the given URL.
+	 *
+	 * @param string $url  The URL to fetch.
+	 * @param array  $args Optional. The arguments to pass to `wp_remote_get()`.
+	 *                     Default empty array.
+	 * @return string The styles on success. Empty string on failure.
+	 */
+	protected function get_remote_styles( $url, array $args = array() ) {
+		// Use a modern user-agent, to get woff2 files.
+		$args['user-agent'] = 'Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:73.0) Gecko/20100101 Firefox/73.0';
+
+		// Get the remote URL contents.
+		$response = wp_safe_remote_get( $url, $args );
+
+		// Early return if the request failed.
+		if ( is_wp_error( $response ) || 200 !== wp_remote_retrieve_response_code( $response ) ) {
+			return '';
+		}
+
+		// Get the response body.
+		return wp_remote_retrieve_body( $response );
+	}
+
+	/**
+	 * Gets the `@font-face` CSS styles for Google Fonts.
+	 *
+	 * This method does the following processing tasks:
+	 *    1. Orchestrates an optimized Google Fonts API URL for each font-family.
+	 *    2. Caches each URL, if not already cached.
+	 *    3. Does a remote request to the Google Fonts API service to fetch the styles.
+	 *    4. Generates the `@font-face` for all its webfonts.
+	 *
+	 * @return string The `@font-face` CSS.
+	 */
+	public function get_css() {
+		$css  = '';
+		$urls = $this->build_collection_api_urls();
+
+		foreach ( $urls as $url ) {
+			$css .= $this->get_cached_remote_styles( 'google_fonts_' . md5( $url ), $url );
```

Comment:
Should we add a `jetpack_` prefix here, so site owners can find them easily, and so we can clean after ourselves easier?

Reply:
Good idea!